manim/mobject/types/vectorized_mobject.py

```diff
@@ -1282,24 +1572,38 @@ def point_from_proportion(self, alpha: float) -> np.ndarray:
             raise ValueError(f"Alpha {alpha} not between 0 and 1.")
 
         self.throw_error_if_no_points()
+        if alpha == 0:
+            return self.points[0]
         if alpha == 1:
             return self.points[-1]
 
-        curves_and_lengths = tuple(self.get_curve_functions_with_lengths())
+        if "piece_curves" not in self.memory:
+            self._init_curve_memory()
+        else:
+            self._update_curve_memory()
 
-        target_length = alpha * sum(length for _, length in curves_and_lengths)
-        current_length = 0
+        lengths = self.memory["piece_curves"]["lengths"]
+        acc_lengths = self.memory["piece_curves"]["acc_lengths"]
+        target_length = alpha * acc_lengths[-1]
 
-        for curve, length in curves_and_lengths:
-            if current_length + length >= target_length:
-                if length != 0:
-                    residue = (target_length - current_length) / length
-                else:
-                    residue = 0
+        num_curves = self.get_num_curves()
 
-                return curve(residue)
+        # Binary search
```

Comment:
No need to implement this manually: use `bisect` from Python's default [bisect](https://docs.python.org/3.8/library/bisect.html) module.

Reply:
Oh nice, the more you know...
I'll check this out!

Reply:
I used `bisect.bisect_left` and it worked as a charm, it was even faster than my manual implementation!